compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -2320,16 +2320,38 @@ private void defineFieldsOfObjectOrRecordTypeDef(BLangTypeDefinition typeDef, Sy
             return;
         }
 
-        if (recordTypeNode.restFieldType == null) {
-            if (recordTypeNode.sealed) {
+        if (recordTypeNode.restFieldType != null) {
+            recordType.restFieldType = symResolver.resolveTypeNode(recordTypeNode.restFieldType, typeDefEnv);
+            return;
+        }
+
+        if (!recordTypeNode.sealed) {
+            recordType.restFieldType = symTable.anydataType;
+            return;
+        }
+
+        for (BLangType typeRef : recordTypeNode.typeRefs) {
+            if (typeRef.type.tag != TypeTags.RECORD) {
+                continue;
+            }
+            BType restFieldType = ((BRecordType) typeRef.type).restFieldType;
+            if (restFieldType == null || restFieldType == symTable.noType) {
+                continue;
+            }
+            if (recordType.restFieldType != null && recordType.restFieldType.tag != restFieldType.tag) {
                 recordType.restFieldType = symTable.noType;
+                dlog.error(recordTypeNode.pos,
+                        DiagnosticErrorCode.CANNOT_USE_TYPE_INCLUSION_WITH_MORE_THAN_ONE_OPEN_RECORD);
```

Comment:
Shouldn't the error be updated now to reflect that they shouldn't be of different types?

Reply:
For example
`cannot use type inclusion with more than one open record with different rest descriptor types`

Reply:
Yes, changed